### PR TITLE
docs: update how-to-use-k8s-with-cri-containerd-and-kata.md

### PR DESCRIPTION
**Pull request by liubin**
Update how-to-use-k8s-with-cri-containerd-and-kata.md to fit the latest
Kubernetes way.
And also changed CNI plugin from flannel to bridge, that will be easy to run.

Fixes: #1325

Signed-off-by: bin <bin@hyper.sh>

### CODE REVIEW

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 1]
```diff
@@ -172,34 +162,48 @@ By default, the cluster will not schedule pods in the master node. To enable mas
...
 ```
 
-## Create an untrusted pod using Kata Containers
+## Create runtime class for Kata Containers
 
 By default, all pods are created with the default runtime configured in CRI containerd plugin.
+From Kubernetes v1.12, user can use [`RuntimeClass`](https://kubernetes.io/docs/concepts/containers/runtime-class/#runtime-class) to specify different runtime for Pods.
+
+```bash
+cat > runtime.yaml <<EOF
```
Comment (jodh-intel) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:171]: Can you add the dollar prompt here:

```suggestion
$ cat > runtime.yaml <<EOF
```

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 2]
```diff
@@ -172,34 +162,48 @@ By default, the cluster will not schedule pods in the master node. To enable mas
...
+apiVersion: node.k8s.io/v1beta1
+kind: RuntimeClass
+metadata:
+  name: kata
+handler: kata
+EOF
+
+sudo -E kubectl apply -f runtime.yaml
```
Comment (jodh-intel) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:179]:
```suggestion
$ sudo -E kubectl apply -f runtime.yaml
```

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 3]
```diff
@@ -125,43 +123,35 @@ $ sudo systemctl daemon-reload
   $ sudo -E kubectl get pods
   ```
 
-## Install a Pod Network
+## Configure Pod Network
 
 A pod network plugin is needed to allow pods to communicate with each other.
+The simplest type of CNI is `bridge`.
+But you can find more about CNI plugins from the [Using `kubeadm` to Create a Cluster](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/#instructions) guide,
```
Comment (jodh-intel) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:130]: s/to Create a Cluster/to create a cluster/

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 4]
```diff
@@ -125,43 +123,35 @@ $ sudo systemctl daemon-reload
...
-          break
-      fi
-
-      sleep 1s
-      ((timeout_dns--))
-   done
-  ```
-
-- Check the pod network is running
+By default the CNI plugins binaries is installed under `/opt/cni/bin` (in package `kubernetes-cni`), you noly need to create a configuration file for CNI plugin.
```
Comment (jodh-intel) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:133]: Typo: "only"

**Review by jodh-intel — COMMENTED** [review]
Thanks @liubin

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md (hunk 3 above)
Comment (liubin) [reply docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:130]: Maybe change to the page title of the target URL will be more suitable: Creating a cluster with `kubeadm`

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 5]
```diff
@@ -125,43 +123,35 @@ $ sudo systemctl daemon-reload
   $ sudo -E kubectl get pods
   ```
 
-## Install a Pod Network
+## Configure Pod Network
 
 A pod network plugin is needed to allow pods to communicate with each other.
+The simplest type of CNI is `bridge`.
+But you can find more about CNI plugins from the [Creating a cluster with `kubeadm`](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/#instructions) guide,
```
Comment (Tim-Zhang) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:130]: Why use `But`?

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 6]
```diff
@@ -125,43 +123,35 @@ $ sudo systemctl daemon-reload
...
+starting from the **Installing a pod network** section.
```
Comment (Tim-Zhang) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:131]: no verb here.

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 7]
```diff
@@ -125,43 +123,35 @@ $ sudo systemctl daemon-reload
...
-          break
-      fi
-
-      sleep 1s
-      ((timeout_dns--))
-   done
-  ```
-
-- Check the pod network is running
+By default the CNI plugins binaries is installed under `/opt/cni/bin` (in package `kubernetes-cni`), you only need to create a configuration file for CNI plugin.
```
Comment (Tim-Zhang) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:133]: CNI plugins binaries -> CNI plugin binaries

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 8]
```diff
@@ -172,34 +162,48 @@ By default, the cluster will not schedule pods in the master node. To enable mas
 $ sudo -E kubectl taint nodes --all node-role.kubernetes.io/master-
 ```
 
-## Create an untrusted pod using Kata Containers
+## Create runtime class for Kata Containers
 
 By default, all pods are created with the default runtime configured in CRI containerd plugin.
+From Kubernetes v1.12, user can use [`RuntimeClass`](https://kubernetes.io/docs/concepts/containers/runtime-class/#runtime-class) to specify different runtime for Pods.
```
Comment (Tim-Zhang) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:168]: user -> users, to specify [a/the] different runtime for Pods

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 9]
```diff
@@ -7,9 +7,10 @@
     * [Configure Kubelet to use containerd](#configure-kubelet-to-use-containerd)
     * [Configure HTTP proxy - OPTIONAL](#configure-http-proxy---optional)
 * [Start Kubernetes](#start-kubernetes)
-* [Install a Pod Network](#install-a-pod-network)
+* [Configure Pod Network](#configure-pod-network)
```
Comment (Tim-Zhang) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:10]: Configure Pod Network -> Configure The Pod Network?

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 10]
```diff
@@ -7,9 +7,10 @@
...
 * [Allow pods to run in the master node](#allow-pods-to-run-in-the-master-node)
-* [Create an untrusted pod using Kata Containers](#create-an-untrusted-pod-using-kata-containers)
+* [Create runtime class for Kata Containers](#create-runtime-class-for-kata-containers)
```
Comment (Tim-Zhang) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:12]: I think it should be `Create a/the runtime class`

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md [hunk 11]
```diff
@@ -125,43 +123,35 @@ $ sudo systemctl daemon-reload
   $ sudo -E kubectl get pods
   ```
 
-## Install a Pod Network
+## Configure Pod Network
```
Comment (Tim-Zhang) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:126]: Configure a/the Pod Network

**Comment by fidencio** [comment]
@liubin, firstly, thanks a lot for working on this one!
Could you address the comments from @Tim-Zhang so we could move forward with this one?

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md (hunk 6 above)
Comment (Tim-Zhang) [inline docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:131]: It's hard to understand this sentence.

**Review by GabyCT — APPROVED** [review]

File: docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md (hunk 6 above)
Comment (fidencio) [reply docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:131]: @liubin, what about the following text:

```
The simplest type of CNI is `bridge`.
But you can find more about CNI plugins from the [Creating a cluster with `kubeadm`](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/#instructions) guide, specifically in the [pod network section](
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#pod-network).
```

Comment (fidencio) [reply docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:131]: @Tim-Zhang, does that clarify the sentence? If so, @liubin, mind to change it?

I really would like to have this merged. :-)

Comment (Tim-Zhang) [reply docs/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md:131]: You can find more about the plugin from the [Creating a cluster with `kubeadm`](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/#instructions) guide.

**Review by Tim-Zhang — APPROVED** [review]

**Pull request merged**